### ISSUE TITLE
V2 Custom Expectation Custom Rendering No Longer Works

### ISSUE THREAD

**Issue by jdimatteo** [fivetran/great_expectations#3174]
**Describe the bug**
V2 custom expectation custom rendering (as documented [here](https://docs.greatexpectations.io/en/latest/reference/spare_parts/data_docs_reference.html#customizing-data-docs)) no longer works.

**To Reproduce**
1. `git clone --branch jdimatteo/reproduce-custom-expectation-rendering-bug git@github.com:jdimatteo/great_expectations.git && cd great_expectations/tests/integration/docusaurus/connecting_to_your_data/database/`
2. `python3.9 -m venv venv0.13.25 && source venv0.13.25/bin/activate`
3. `pip install -r requirements-0.13.25.txt`
4. `python sqlite_python_example_with_v2_api.py`

This should run without error as it does with GE version 0.12.7 (see details below), however instead fails as shown in this AssertionError:

```
Traceback (most recent call last):
  File "/home/jdimatteo/dev/great_expectations_fork_1/tests/integration/docusaurus/connecting_to_your_data/database/sqlite_python_example_with_v2_api.py", line 161, in <module>
    main()
  File "/home/jdimatteo/dev/great_expectations_fork_1/tests/integration/docusaurus/connecting_to_your_data/database/sqlite_python_example_with_v2_api.py", line 158, in main
    assert_custom_rendering_present_in_html()
  File "/home/jdimatteo/dev/great_expectations_fork_1/tests/integration/docusaurus/connecting_to_your_data/database/sqlite_python_example_with_v2_api.py", line 141, in assert_custom_rendering_present_in_html
    assert "**{" not in html_file_contents
AssertionError
```

This `AssertionError` corresponds to the missing custom rendering as shown here:

![image](https://user-images.githubusercontent.com/3752732/128114821-208db9d3-741b-46eb-bbd2-b4973bc62095.png)

**Expected behavior**

Custom rendering should work as it did in prior GE version 0.12.7:

1. `git clone --branch jdimatteo/reproduce-custom-expectation-rendering-bug git@github.com:jdimatteo/great_expectations.git && cd great_expectations/tests/integration/docusaurus/connecting_to_your_data/database/`
2. `python3.9 -m venv venv0.12.7 && source venv0.12.7/bin/activate`
3. `pip install -r requirements-0.12.7.txt`
4. `python sqlite_python_example_with_v2_api.py`

This runs without error and produces expected custom rendering:

![image](https://user-images.githubusercontent.com/3752732/128114904-f6118a7d-931a-4bf9-aa3c-ad247b3c0f9a.png)

**Environment (please complete the following information):**
 - Operating System: Linux
 - Great Expectations Version: 0.13.25

**Additional context**

I understand that the V2 API is going to be deprecated at some point, however it isn't deprecated yet and the V3 API isn't ready for all use cases because it is documented as experimental [here](https://docs.greatexpectations.io/en/latest/guides/how_to_guides/migrating_versions.html) and there are several bugs preventing usage that might not be fixed for several weeks (e.g. https://github.com/great-expectations/great_expectations/issues/3095). Using an old version of Great Expectations is also not an option, because doing so requires using older versions of other libraries as well such as urllib3 with security vulnerabilities including https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-33503.

See also discussion at https://greatexpectationstalk.slack.com/archives/CUTCNHN82/p1628046306055600.

**Comment by jdimatteo** [comment]
I'm working on a PR to fix this

**Comment by talagluck** [comment]
Hi @jdimatteo - thanks for raising the issue, and thanks for working on the PR! I'll keep an eye out for it over the next few days. Please let me know if I can be of any assistance.

**Comment by jdimatteo** [comment]
Do V3 style custom expectations work when using the V2 API?

https://docs.greatexpectations.io/en/latest/guides/how_to_guides/creating_and_editing_expectations/how_to_create_custom_expectations.html

I'll probably test this soon if nobody knows off hand.

**Comment by talagluck** [comment]
Hi @jdimatteo - no, custom expectations were a major impetus for the V3 API and are mutually exclusive from V2 expectations. Expectations in V3 are each in their own file, whereas in V2 they all reside in backend-specific dataset modules (e.g. `pandas_dataset.py`)

**Comment by jdimatteo** [comment]
Thanks @talagluck for the info! I've got this partially working locally and will submit a PR soon hopefully.

**Comment by jdimatteo** [comment]
@talagluck please see the PR with the fix at #3179

**Comment by jdimatteo** [comment]
Note: a full working end to end example testing the PR is at https://github.com/great-expectations/great_expectations/blob/3d3cea6c052e87b71434e26b874919531c12e002/tests/integration/docusaurus/connecting_to_your_data/database/custom_rendering_with_v2_api_example.py.
